configure

```diff
@@ -0,0 +1,360 @@
+#!/bin/sh
+### Custom configure script of Dinit.
+
```

Comment:
Hm. about `set -e`, I created a error system as `error()` and `sub_error()` function, so when we catch a error, print a error message and exit immediately. (except before `error()` and `sub_error()` ) so I think I don't use this thing.
About `set -u`, My script checks a variable exist or not, `set -u` broke this thing.
```
Info: Checking C++ Compiler...
./configure: 223: CXX: parameter not set
```
but `$CXX` is not defined in some cases. so if we need to use `set -u`, I need to change some basic things.

Reply:
`set -e` is good practice for catching cases where you forgot to explicitly check. I can see that you have put explicit checks for a lot of things and these will continue to work even if you use `set -e`.

If using `set -u` and you want to refer to a variable that may not be set, you can use `${VARNAME:-}`.

Reply:
I do it. but seems like FreeBSD CIs are fail. I figure it out.

Reply:
FreeBSD CI has fixed.

Reply:
Keep in mind https://mywiki.wooledge.org/BashFAQ/105

`set -e` often doesn't do what people expect.

Reply:
I'm aware of the issues with `set -e`, but using it still gives better behaviour than not using it in my opinion.

Reply:
From [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/):
> -e
    When this option is on, when any command fails (for any of the reasons listed in [Consequences of Shell Errors](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_08_01) or by returning an [exit](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#exit) status greater than zero), the shell immediately shall exit, as if by executing the exit special built-in utility with no arguments, with the following exceptions:
>  * The failure of any individual command in a multi-command pipeline shall not cause the shell to exit. Only the failure of the pipeline itself shall be considered.
>  * The -e setting shall be ignored when executing the compound list following the while, until, if, or elif reserved word, a pipeline beginning with the ! reserved word, or any command of an AND-OR list other than the last.
> * If the exit status of a compound command other than a subshell command was the result of a failure while -e was being ignored, then -e shall not apply to this command.
> * This requirement applies to the shell environment and each subshell environment separately. For example, in:
> ` set -e; (false; echo one) | cat; echo two`
> the [false](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/false.html) command causes the subshell to exit without executing echo one; however, echo two is executed because the exit status of the pipeline (false; echo one) | cat is zero.

According to this, I prefer to use `set -e`.